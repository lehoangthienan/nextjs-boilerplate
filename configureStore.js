import { createStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { composeWithDevTools } from 'redux-devtools-extension'
import { fromJS } from 'immutable'

import rootReducer from './reducers'
import rootSaga from './sagas'

const sagaMiddleware = createSagaMiddleware()

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    return composeWithDevTools(applyMiddleware(...middleware))
  }
  return applyMiddleware(...middleware)
}

export default function configureStore(initialState = {}) {
  const store = createStore(
    rootReducer,
    fromJS(initialState),
    bindMiddleware([sagaMiddleware])
  )

  /**
   * next-redux-saga depends on `runSagaTask` and `sagaTask` being attached to the store.
   *
   *   `runSagaTask` is used to rerun the rootSaga on the client when in sync mode (default)
   *   `sagaTask` is used to await the rootSaga task before sending results to the client
   *
   */

  store.runSagaTask = () => {
    store.sagaTask = sagaMiddleware.run(rootSaga)
  }

  // run the rootSaga initially
  store.runSagaTask()
  return store
}
