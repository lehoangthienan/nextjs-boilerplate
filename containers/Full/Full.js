import React, { Component } from 'react'
import PropTypes from 'prop-types'
import dynamic from 'next/dynamic'
import { connect } from 'react-redux'
import loGet from 'lodash/get'
import Cookie from 'js-cookie'

import AppActions from '../../redux/appRedux'
import { connectSocket, disConnectSocket } from '../../services/socketService'

const Header = dynamic({ loader: () => import('../../components/Header') })
const Footer = dynamic({ loader: () => import('../../components/Footer') })

class Full extends Component {
  componentDidMount() {
    const {
      startupWorkingFlow,
      history,
      dispatch,
    } = this.props
    startupWorkingFlow(history)

    const token = Cookie.get('token') || ''
    if (token) {
      const user = Cookie.get('user') || {}
      connectSocket(token, dispatch, loGet(user, ['_id'], ''))
    }
  }

  componentWillUnmount() {
    disConnectSocket()
  }

  render() {
    const { children } = this.props
    return (
      <div>
        <Header />
        {children}
        <Footer />
      </div>
    )
  }
}

Full.propTypes = {
  startupWorkingFlow: PropTypes.func,
  history: PropTypes.object,
  children: PropTypes.any,
  dispatch: PropTypes.any,
}

const mapStateToProps = (state) => ({
  isReady: state.toJS().app.isReady,
})

export const mapDispatchToProps = (dispatch) => ({
  dispatch,
  startupWorkingFlow: (history) => {
    dispatch(AppActions.startupWorkingFlow(history)) //eslint-disable-line
  },
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Full)
