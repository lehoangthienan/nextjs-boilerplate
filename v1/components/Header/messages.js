import { defineMessages } from 'react-intl'

export const scope = 'app.components.Header'

export default defineMessages({
  vi: {
    id: `${scope}.vi`,
    defaultMessage: 'Viet Nam',
  },
  en: {
    id: `${scope}.en`,
    defaultMessage: 'English',
  },
})
