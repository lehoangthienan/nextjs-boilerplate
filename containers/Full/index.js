import Full from './Full';

export default Full;
