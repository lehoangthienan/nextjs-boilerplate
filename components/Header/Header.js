import React, { Component } from 'react';
import { Select } from 'antd';
import { FormattedMessage, injectIntl } from 'react-intl';
import Cookie from 'js-cookie';

import messages from './messages';

const { Option } = Select;

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      locale: Cookie.get('locale') || 'vi',
    };
  }

  handleChange = (locale) => {
    Cookie.set('locale', locale);
    window.location.reload(true);
    this.setState({ locale });
  }

  render() {
    const { locale } = this.state;
    return (
      <div>
        <Select value={locale} style={{ width: 120 }} onChange={this.handleChange}>
          <Option value="vi"><FormattedMessage {...messages.vi} /></Option>
          <Option value="en"><FormattedMessage {...messages.en} /></Option>
        </Select>
      </div>
    );
  }
}

export default (injectIntl(Header));
