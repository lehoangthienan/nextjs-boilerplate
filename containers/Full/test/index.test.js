import React from 'react'
import renderer from 'react-test-renderer'
import preloadAll from 'jest-next-dynamic'
import { Provider as ReduxProvider } from 'react-redux'
import { createIntl, createIntlCache, RawIntlProvider, defineMessages } from 'react-intl'

import configureStore from '../../../configureStore'
import { scope } from '../../../components/Header/messages'
import Full from '../index'

describe('<Full />', () => {
  let store
  let intl
  let messages
  
  beforeAll(async () => {
    const cache = createIntlCache()
    store = configureStore({})
    intl = createIntl(
      {
        locale: 'en',
        messages,
      },
      cache
    )
  
    messages = defineMessages({
      vi: {
        id: `${scope}.vi`,
        defaultMessage: 'Viet Nam',
      },
      en: {
        id: `${scope}.en`,
        defaultMessage: 'English',
      },
    })

    await preloadAll()
  })
    
  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <ReduxProvider store={store}>
          <RawIntlProvider value={intl}>
            <Full />
          </RawIntlProvider>
        </ReduxProvider>,
      )
      .toJSON()

    expect(renderedComponent).toMatchSnapshot()
  })
})
