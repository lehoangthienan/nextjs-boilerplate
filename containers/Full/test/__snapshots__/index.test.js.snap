// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Full /> should render and match the snapshot 1`] = `
<div>
  <div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 120,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-controls="test-uuid"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
            ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title=""
          >
            Viet Nam
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <i
            aria-label="icon: down"
            className="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
  </div>
  <div
    className="lazyload-placeholder"
    style={
      Object {
        "height": 400,
      }
    }
  />
</div>
`;
